src/Queue.php

```diff
@@ -429,20 +429,23 @@ private function ensureIndex(array $index)
                 //so we use any generated name, and then find the right spec after we have called,
                 //and just go with that name.
                 try {
-                    $this->collection->ensureIndex($index, ['name' => $name, 'background' => true]);
-                } catch (\MongoException $e) {
+                    $this->collection->createIndex($index, ['name' => $name, 'background' => true]);
+                } catch (\MongoDB\Exception\Exception $e) {
                     //this happens when the name was too long, let continue
                 }
 
-                foreach ($this->collection->getIndexInfo() as $existingIndex) {
+                foreach ($this->collection->listIndexes() as $existingIndex) {
                     if ($existingIndex['key'] === $index) {
                         return;
                     }
                 }
+
+                //@codeCoverageIgnoreStart
             }
         }
 
         throw new \Exception('couldnt create index after 5 attempts');
+        //@codeCoverageIgnoreEnd
```

Comment:
You could write a test that covers this if you passed in a collection mock to the constructor that threw a `\MongoDB\Exception\Exception` each time `createIndex` was called.